### PR TITLE
Simplify bugref counting with map

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by asdil12** [comment]
Apart from that being much harder to read and understand - especially without brackets:
Is it even possible to increase the number of elements within a map - i.e. deal with comments that contain multiple bugreferences?

**Comment by kraih** [comment]
@asdil12 Absolutely, that's an important feature of `map`.
```
$ perl -E 'say join ", ", map { $_, $_, $_ } 1, 2, 3'
1, 1, 1, 2, 2, 2, 3, 3, 3
```

File: lib/OpenQA/Schema/ResultSet/Comments.pm [hunk 1]
```diff
@@ -38,12 +38,7 @@ sub compute_bugref_count {
     my ($self) = @_;
 
     my $comments = $self->search({-not => {job_id => undef}});
-    my %bugrefs;
-    for my $comment ($comments->all) {
-        for my $bug (@{find_bugrefs($comment->text)}) {
-            $bugrefs{$bug} += 1;
-        }
-    }
+    my %bugrefs = map { $_ => 1 } map { @{find_bugrefs($_->text)} } $comments->all;
```
Comment (kraih) [inline lib/OpenQA/Schema/ResultSet/Comments.pm]: Wish we could depend on a newer version of Perl. This is one of the cases where i would really like postderef. Maybe some day.
```perl
my %bugrefs = map { $_ => 1 } map { find_bugrefs($_->text)->@* } $comments->all;
```

Comment (Martchus) [inline lib/OpenQA/Schema/ResultSet/Comments.pm]: It looks like it now doesn't count the bugrefs anymore and only returns whether they exist at all. That might be sufficient but you should rename the function name then.

Comment (okurz) [reply lib/OpenQA/Schema/ResultSet/Comments.pm]: good point. Renamed to "referenced_bugs".

**Pull request merged**